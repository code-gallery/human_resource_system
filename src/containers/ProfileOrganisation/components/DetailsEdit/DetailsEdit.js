import React from 'react'
import PropTypes from 'prop-types'
import _isNil from 'lodash/isNil'
import Autocomplete from 'react-google-autocomplete'
import classNames from 'classnames'
import Form from 'components/Abstract/Form'
import ContentCard from 'components/ContentCard'
import Input from 'components/FormElements/Input'
import Select from 'react-select'
import './style.css'

class DetailsEdit extends Form {
  constructor(props) {
    super(props)
    const { organisation } = props
    this.state = {
      name: organisation.name,
      industries: organisation.industries,
      year_founded: organisation.year_founded,
      company_size: organisation.company_size
    }
  }

  onChangeInput(event) {
    super.onChangeInput(event)
    this.props.onChangeInput(event)
  }

  render() {
    const { reference, organisation, errors } = this.props
    const { town, country } = organisation
    const { name, industries, year_founded, company_size } = this.state
    const industriesRef = (reference) ? reference.industries : []
    const { organisationSize } = reference
    const locationPlaceholder =
      (_isNil(town) || town === '') &&
      (_isNil(country) || country === '') ?
        'Enter your address' : `${town}, ${country}`

    return (
      <ContentCard title="Details" className="DetailsEdit">
        <form>
          <Input
            className={classNames({
              'has-error': errors.indexOf('name') !== -1
            })}
            label="Name"
            name="name"
            value={name}
            placeholder="Company name"
            onChange={this.onChangeInput}
          />
          <div
            className={classNames('form-group', {
              'has-error': errors.indexOf('company_size') !== -1
            })}
          >
            <label>Size</label>
            <Select
              name="select-content"
              options={organisationSize}
              labelKey="text"
              valueKey="text"
              value={company_size}
              onChange={(value) => {
                this.onChangeInput({
                  target: {
                    name: 'company_size',
                    value
                  }
                })
              }}
            />
          </div>
          <div
            className={classNames('form-group', {
              'has-error': errors.indexOf('town') !== -1 || errors.indexOf('country') !== -1
            })}
          >
            <label>Location</label>
            <Autocomplete
              onPlaceSelected={(value) => {
                this.onChangeInput({
                  target: {
                    name: 'location',
                    value
                  }
                })
              }}
              placeholder={locationPlaceholder}
              types={[ '(cities)' ]}
            />
          </div>
          <div
            className={classNames('form-group', {
              'has-error': errors.indexOf('industries') !== -1
            })}
          >
            <label>Industry</label>
            <Select
              name="select-content"
              options={industriesRef}
              labelKey="text"
              valueKey="text"
              value={industries}
              onChange={(value) => {
                this.onChangeInput({
                  target: {
                    name: 'industries',
                    value
                  }
                })
              }}
            />
          </div>
          <Input
            className={classNames({
              'has-error': errors.indexOf('year_founded') !== -1
            })}
            label="Year Founded"
            name="year_founded"
            value={year_founded}
            placeholder="YYYY"
            onChange={this.onChangeInput}
          />
        </form>
      </ContentCard>
    )
  }
}

DetailsEdit.propTypes = {
  organisation: PropTypes.shape({
    name: PropTypes.string,
    town: PropTypes.string,
    industries: PropTypes.string,
    company_size: PropTypes.string,
    year_founded: PropTypes.string
  }),
  onChangeInput: PropTypes.func.isRequired,
  reference: PropTypes.shape({
    awards: PropTypes.array,
    degrees: PropTypes.array,
    industries: PropTypes.array
  }),
  errors: PropTypes.array.isRequired
}

export default DetailsEdit
