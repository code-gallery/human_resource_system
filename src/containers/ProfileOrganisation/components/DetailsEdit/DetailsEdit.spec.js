import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import DetailsEdit from './index'

describe('containers/ProfileOrganisation/components/DetailsEdit', () => {
  const commonProps = {
    reference: {
      industries: [
        { text: 'Accounting' },
        { text: 'Banking' },
        { text: 'IT' }
      ],
      organisationSize: [
        { text: '1-10 employees' },
        { text: '11-50 employees' },
        { text: '51-200 employees' }
      ]
    },
    onChangeInput: jest.fn()
  }

  describe('when most fields are empty and no errors', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        organisation: {
          name: 'Applied Blockchain'
        },
        errors: []
      }
      const tree = renderer.create(
        <DetailsEdit {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when all fields exists and no errors', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        organisation: {
          name: 'Applied Blockchain',
          industries: 'IT',
          company_size: '11-50 employees',
          town: 'London',
          country: 'UK',
          year_founded: '2015'
        },
        errors: []
      }
      const tree = renderer.create(
        <DetailsEdit {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when most fields are empty with errors', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        organisation: {
          name: 'Applied Blockchain',
          town: 'London'
        },
        errors: [ 'location', 'company_size', 'country', 'industries', 'year_founded' ]
      }
      const tree = renderer.create(
        <DetailsEdit {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('<DetailsEdit />', () => {
    let wrapper
    let props
    beforeEach(() => {
      props = {
        ...commonProps,
        organisation: {
          name: 'Applied Blockchain',
          industries: 'IT',
          company_size: '11-50 employees',
          town: 'London',
          country: 'UK',
          year_founded: '2015'
        },
        errors: []
      }
      wrapper = shallow(
        <DetailsEdit {...props} />
      )
    })

    describe('onChangeInput', () => {
      it('calls props.onChangeInput', () => {
        const event = { target: { type: 'text', value: 'hello' } }
        wrapper.instance().onChangeInput(event)
        expect(props.onChangeInput).toHaveBeenCalledWith(event)
      })
    })
  })
})
