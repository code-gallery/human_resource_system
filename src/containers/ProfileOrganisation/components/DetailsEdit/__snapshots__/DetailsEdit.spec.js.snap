// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/ProfileOrganisation/components/DetailsEdit when all fields exists and no errors renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 DetailsEdit"
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      Details
    </h3>
    <form>
      <div
        className="form-group "
      >
        <label>
          Name
        </label>
        <input
          name="name"
          onChange={[Function]}
          placeholder="Company name"
          required={false}
          type="text"
          value="Applied Blockchain"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Size
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
          style={undefined}
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value="11-50 employees"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-4--value"
            >
              <div
                className="Select-value"
                style={undefined}
                title={undefined}
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-4--value-item"
                  role="option"
                >
                  11-50 employees
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-4--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label>
          Location
        </label>
        <div>
          <input
            placeholder="London, UK"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label>
          Industry
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
          style={undefined}
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value="IT"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-5--value"
            >
              <div
                className="Select-value"
                style={undefined}
                title={undefined}
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-5--value-item"
                  role="option"
                >
                  IT
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-5--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-group "
      >
        <label>
          Year Founded
        </label>
        <input
          name="year_founded"
          onChange={[Function]}
          placeholder="YYYY"
          required={false}
          type="text"
          value="2015"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`containers/ProfileOrganisation/components/DetailsEdit when most fields are empty and no errors renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 DetailsEdit"
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      Details
    </h3>
    <form>
      <div
        className="form-group "
      >
        <label>
          Name
        </label>
        <input
          name="name"
          onChange={[Function]}
          placeholder="Company name"
          required={false}
          type="text"
          value="Applied Blockchain"
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Size
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-2--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-2--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label>
          Location
        </label>
        <div>
          <input
            placeholder="Enter your address"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label>
          Industry
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-3--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-3--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-group "
      >
        <label>
          Year Founded
        </label>
        <input
          name="year_founded"
          onChange={[Function]}
          placeholder="YYYY"
          required={false}
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`containers/ProfileOrganisation/components/DetailsEdit when most fields are empty with errors renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 DetailsEdit"
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      Details
    </h3>
    <form>
      <div
        className="form-group "
      >
        <label>
          Name
        </label>
        <input
          name="name"
          onChange={[Function]}
          placeholder="Company name"
          required={false}
          type="text"
          value="Applied Blockchain"
        />
      </div>
      <div
        className="form-group has-error"
      >
        <label>
          Size
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-6--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-6--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-group has-error"
      >
        <label>
          Location
        </label>
        <div>
          <input
            placeholder="London, undefined"
          />
        </div>
      </div>
      <div
        className="form-group has-error"
      >
        <label>
          Industry
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-7--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-7--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-group has-error"
      >
        <label>
          Year Founded
        </label>
        <input
          name="year_founded"
          onChange={[Function]}
          placeholder="YYYY"
          required={false}
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
</div>
`;
